### PR TITLE
Update Docs.MS ToC sync to handle discovered fusionconfig limitations

### PR DESCRIPTION
**Pull request by scbedd**
Our documents are presented on docs.ms in two main views.

- `Latest`
- `Preview`
- ...and `Legacy` exists but we don't really care to actively update it

In the previous iteration of `sync_tocs`, the `preview` ToC would fall back to the `GA` readme if a preview version didn't exist. This is actually ok, except for the fact that you cannot refer to the same single document across multiple monikers. It _has_ to be differentiable. If it's NOT, then the way that the fusion ToC (read, merged across monikers) is rendered goes sideways on UI side.

This update handles the issue automatically.

@Azure/azure-sdk-eng

### CODE REVIEW

**Comment by scbedd** [comment]
@Azure/azure-sdk-eng an someone take a look and rubber stamp so I can merge this?

File: scripts/python/docs-automation/sync_tocs.py [hunk 1]
```diff
@@ -31,19 +31,20 @@ def filter_children(targeted_ns_list, known_namespaces):
...
+    input_string = toc_dict["href"]
 
     resolvable_path = os.path.join(repo_location, input_string.replace("~/", ""))
-    possible_target_readme = os.path.splitext(resolvable_path)[0] + "-" + suffix
+    possible_target_readme = os.path.splitext(resolvable_path)[0] + suffix
 
     if os.path.exists(possible_target_readme):
+        toc_dict["href"] = input_string.replace(".md", suffix)
+    else:
+        toc_dict.pop("href")
```
Comment (weshaggard) [inline scripts/python/docs-automation/sync_tocs.py:44]: I don't have enough context here but if you remove the href does this mean that there will not be a readme link provided in the ToC?

Comment (scbedd) [reply scripts/python/docs-automation/sync_tocs.py:44]: Yes. This is intentional. I'd rather show NO readme than the wrong readme.

Comment (scbedd) [reply scripts/python/docs-automation/sync_tocs.py:44]: Not only this, but if the same readme is requested from two different ToCs (latest and preview), the entire node ceases to render.

**Pull request merged**